simulator/func_sim/operation.h

```diff
@@ -163,18 +167,13 @@ class Datapath : public Operation
     using RegisterUInt = T;
     using RegisterSInt = sign_t<RegisterUInt>;
 
-    void set_v_src( const T& value, size_t index)
-    {
-        if ( index == 0)
-            v_src1 = value;
-        else
-            v_src2 = value;
-    }
+    void set_v_src(  const T &value, size_t index) { v_src[index] = value; }
+    void set_v_dst( const T &value, size_t index) { v_dst[index] = value; }
```

Comment:
Please use v_src.at( index) to handle out-of-range accesses

Reply:
Should we use `at` for get methods as well?

Reply:
Sure.

Reply:
`at` methods have to be used if index value is not a compile-time constant